GRF/GRF.cs

```diff
@@ -1,204 +1,246 @@
-            var bodySize = streamReader.ReadInt32();
-
-            var compressedBody = streamReader.ReadBytes( compressedBodySize );
-            var bodyData = ZlibStream.UncompressBuffer( compressedBody );
-
-            var bodyStream = new MemoryStream( bodyData );
-            var bodyReader = new BinaryReader( bodyStream );
-
-            for( int i = 0; i < fileCount; i++ )
-            {
-                var fileName = string.Empty;
-                char currentChar;
-                while( ( currentChar = (char)bodyReader.ReadByte() ) != 0 )
-                {
-                    fileName += currentChar;
-                }
-
-                var compressedFileSize = bodyReader.ReadInt32();
-                var compressedFileSizeAligned = bodyReader.ReadInt32();
-                var uncompressedFileSize = bodyReader.ReadInt32();
-                var fileFlags = (FileFlag)bodyReader.ReadByte();
-                var fileDataOffset = bodyReader.ReadInt32();
-
-                // skip directories and files with zero size
-                if( !fileFlags.HasFlag( FileFlag.File ) || uncompressedFileSize == 0 )
-                    continue;
-
-                streamReader.BaseStream.Seek( GrfHeaderSize + fileDataOffset, SeekOrigin.Begin );
-
-                Entries.Add(
-                    fileName,
-                    new GrfEntry(
-                        streamReader.ReadBytes( compressedFileSizeAligned ),
-                        fileName,
-                        compressedFileSize,
-                        uncompressedFileSize,
-                        fileFlags ) );
-            }
-
-            bodyReader.Close();
-        }
-
-        private string DecodeFileName( Span<byte> encodedName )
-        {
-            for( int i = 0; i < encodedName.Length; i++ )
-            {
-                // swap nibbles
-                encodedName[i] = (byte)( ( encodedName[i] & 0x0F ) << 4 | ( encodedName[i] & 0xF0 ) >> 4 );
-            }
-            for( int i = 0; i < encodedName.Length / DataEncryptionStandard.BlockSize; i++ )
-            {
-                DataEncryptionStandard.DecryptBlock( encodedName.Slice(
-                    i * DataEncryptionStandard.BlockSize,
-                    DataEncryptionStandard.BlockSize ) );
-            }
-
-            string fileName = string.Empty;
-            for( int i = 0; i < encodedName.Length; i++ )
-            {
-                if( (char)encodedName[i] == 0 )
-                    break;
-
-                fileName += (char)encodedName[i];
-            }
-
-            return fileName;
-        }
-
-        private bool IsFullEncrypted( string fileName )
-        {
-            string[] extensions = { ".gnd", ".gat", ".act", ".str" };
-            foreach( var extension in extensions )
-            {
-                if( fileName.EndsWith( extension ) )
-                    return false;
-            }
-            return true;
-        }
-    }
-}
+﻿using System;

```

Comment:
There are too many differences, I think you broke the line endings.

Reply:
I've broken the code the line endings, only notice now sorry.

Reply:
Ok. I fixed the line feed, you can see all changes over GRF.cs file =]

Reply:
Thank you :smiley: